### DISCUSSION TITLE
MiniCssExtractPlugin and HtmlWebpackPlugin: How to add CSP nonce

### DISCUSSION THREAD

**Discussion by bozdoz** [webpack/webpack#16387] [Q&A]
I'm not sure how to get HtmlWebpackPlugin to:
1. generate a meta tag for CSP
2. add a nonce to style tags (is this todo with MiniCssExtractPlugin?)
3. add a nonce to script tags

I have tried adding `__webpack_nonce__ = 'c29tZSBjb29sIHN0cmluZyB3aWxsIHBvcCB1cCAxMjM='` as described here: https://webpack.js.org/guides/csp/

That does not seem to affect anything at all.

Any help would be greatly appreciated.

**Comment by alexander-akait** [comment]
Somebody should send a PR with supporting them (feel free to do it), `HtmlWebpackPlugin` can get them from stats
